src/Build/BackEnd/Components/Communications/NodeProviderInProc.cs

```diff
@@ -367,7 +367,7 @@ private bool InstantiateNode(INodePacketFactory factory)
 
             int connectionTimeout = CommunicationsUtilities.NodeConnectionTimeout;
             bool connected = _endpointConnectedEvent.WaitOne(connectionTimeout);
-            ErrorUtilities.VerifyThrow(connected, "In-proc node failed to start up within {0}ms", connectionTimeout);
+            ErrorUtilities.VerifyThrow(connected, "In-proc node failed to start up within {0}ms. That timeout is overriden from its default by the environment variable MSBUILDNODECONNECTIONTIMEOUT. Setting that to a higher value may help.", connectionTimeout);
```

Comment:
Can you distinguish between the default and the overridden case? Now it will fire if the (normal) timeout is too short.